File: lc_soln/0121_best_time_to_buy_and_sell_stock/solution.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy = float("inf")
        result = 0
        for p in prices:
            if p < buy:
                buy = p
            elif p - buy > result:
                result = max(result, p - buy)
        return result

File: lc_soln/0121_best_time_to_buy_and_sell_stock/solution.cs
public class Solution
{
    public int MaxProfit(int[] prices)
    {
        int result = 0;
        int buy = int.MaxValue;
        foreach (int price in prices)
        {
            if (price < buy)
            {
                buy = price;
            }
            else
            {
                result = int.Max(result, price - buy);
            }
        }
        return result;
    }
}

